### ISSUE TITLE
<select> ignoring selected when ngModel is object

### ISSUE THREAD

**Issue by gp187** [angular/angular#14399]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
The `NgModel` in the `<select>` updates the data correctly (as object), but when loading the data and placing a `[attr.selected]="true"` it doesn't select that value from options

**Expected behavior**
Loading the form with data, if a `selected` flag is on a `<option>` inside a `<select>` I'd expect that to be selected.

**Minimal reproduction of the problem with instructions**
```typescript
import {Component, NgModule} from '@angular/core'
import {BrowserModule } from '@angular/platform-browser'
import { FormsModule } from '@angular/forms';

@Component({
  selector: 'my-app',
  template: `
    <select [(ngModel)]="data.currency" name="currency" id="currency">
        <option *ngFor="let c of currs" [ngValue]="c" [attr.selected]="c.value == data.currency.value ? true : null">{{ c.name  }}</option>
    </select>
  `,
})
export class App {

  currs = [
    {name: "EURO", value: "EUR"},
    {name: "Dollar", value: "DOLLAR"},  
    {name: "British Pound", value: "Pound"}
  ];
  
  data: {
    currency: currs[2]
  };

}

@NgModule({
  imports: [ BrowserModule, FormsModule ],
  declarations: [ App ],
  bootstrap: [ App ]
})
export class AppModule {}
```

**What is the motivation / use case for changing the behavior?**
Saving `objects` from  `<select>` via `ngModel` is very helpful and prevents an extra lookup when using the data. But when returning to the form, the correct option doesn't show. It just shows blank `<select>` even if the data is accurate and the model is set correctly. 

**Please tell us about your environment:**
Ubuntu 16.04
Chrome 56.0

* **Angular version:** 2.4.3

* **Browser:** [all | Chrome 56 | Firefox 51 |
 
* **Language:** [all | TypeScript 2.1 | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` = v6.9.4

#### FIX
https://github.com/angular/angular/blob/2.4.7/modules/%40angular/forms/src/directives/select_control_value_accessor.ts#L113
Setting a 2nd lookup for the `selected` attr and return the value would do the trick. 
I think

**Comment by DzmitryShylovich** [comment]
Angular ignored `selected` attr for a reason. See https://github.com/angular/material2/issues/2250#issuecomment-274904991

btw nice issue title. very descriptive :)

**Comment by gp187** [comment]
Right.. :) new title. 
Would be good if the value is `false` to recheck for selected attribute. 
Then it wouldn't be duplicative

Or when NgModel is object

**Comment by DzmitryShylovich** [comment]
With `ngValue` you can provide `compareWith` function https://github.com/angular/angular/pull/13349 to compare options. This feature is on;y available in 4.0

**Comment by gp187** [comment]
Thanks
